modules/graphics/waybar.config.nix

```diff
@@ -0,0 +1,335 @@
+    path,
+    ...
+  }: ''
+    "custom/${name}": {
+              "format": " ",
+              "on-click": "${builtins.replaceStrings ["\""] ["\\\\\\\""] path}",
+              "tooltip": false
+    },
+  '';
+  mkLauncherModule = {name, ...}: ''"custom/${name}", '';
+
+  /*
+  Generate launchers to be used in /etc/waybar/config
+
+  Type: mkLaunchers :: [{name, path, icon}] -> string
+
+  */
+  mkLauncherStyles = lib.concatMapStrings mkLauncherStyle;
+  mkLauncherConfigs = lib.concatMapStrings mkLauncherConfig;
+  mkLauncherPaddings = lib.concatMapStrings mkLauncherPadding;
+  mkLauncherModules = lib.concatMapStrings mkLauncherModule;
+
+  defaultLauncher = [
+    # Keep weston-terminal launcher always enabled explicitly since if someone adds
+    # a launcher on the panel, the launcher will replace weston-terminal launcher.
+    {
+      name = "terminal";
+      path = "${pkgs.weston}/bin/weston-terminal";
+      icon = "${pkgs.weston}/share/weston/icon_terminal.png";
+    }
+  ];
+
+  wifi-signal-strength = pkgs.callPackage ../../packages/wifi-signal-strength {};
+in {
+  config = lib.mkIf cfg.enable {
+    ghaf.graphics.launchers = defaultLauncher;
+    environment.etc."waybar/config" = {
+      text =
+        # Modified from default waybar configuration file https://github.com/Alexays/Waybar/blob/master/resources/config
+        ''
+          {
+            "height": 30, // Waybar height (to be removed for auto height)
+            "spacing": 4, // Gaps between modules (4px)
+            "modules-left": [ ''
+        + mkLauncherModules config.ghaf.graphics.launchers
+        + ''          ],
+                      "modules-center": ["sway/window"],
+                      "modules-right": ["pulseaudio", "custom/network1", "backlight", "battery", "clock", "tray"],
+                      "keyboard-state": {
+                          "numlock": true,
+                          "capslock": true,
+                          "format": "{name} {icon}",
+                          "format-icons": {
+                              "locked": "",
+                              "unlocked": ""
+                          }
+                      },
+                      "tray": {
+                          // "icon-size": 21,
+                          "spacing": 10
+                      },
+                      "clock": {
+                          "timezone": "Europe/Helsinki",
+                          "tooltip-format": "<big>{:%Y %B}</big>\n<tt><small>{calendar}</small></tt>",
+                          "format-alt": "{:%Y-%m-%d}"
+                      },
+                      "backlight": {
+                          // "device": "acpi_video1",
+                          "format": "{percent}% {icon}",
+                          "format-icons": ["", "", "", "", "", "", "", "", ""]
+                      },
+                      "battery": {
+                          "states": {
+                              // "good": 95,
```

Comment:
Why we comment out `good`?

Reply:
I expected that it was working with builtin module but I figured out they don't change when testing then removed both `good` and `warning`.